### PR TITLE
[tests-only][full-ci] web e2e with multiple storage-users services

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Running web e2e tests on ocis with multi storage-users services.

Screenshot from 2024-11-19 11-26-01

Two storage-users services are setup with shared storage volume (`/root/.ocis` in CI) and `nats-js-kv` cache store to make them work with ocis server running without storage-users service (without external load balancer).
```
"OCIS_CACHE_STORE": "nats-js-kv",
"OCIS_CACHE_STORE_NODES": "ocis-server:9233",
```

>[!NOTE]
>- ~Currently, we run this pipeline on every PRs.~ The pipeline will run on `full-ci` PRs and nightly
~>- e2e `smoke` test suite is skipped because large file upload test is not passing with drone setup (but works with local docker-compose setup)~ fixed with cors config (https://github.com/owncloud/ocis/issues/7614#issuecomment-2485309180)

## Related Issue
- Resolves https://github.com/owncloud/ocis/issues/7614

## How Has This Been Tested?
- test environment: :robot: 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by micbar** [comment]
@saw-jan nice idea!

@dragonchaser @butonic couldn‘t we do that without nginx?

**Comment by saw-jan** [comment]
>couldn‘t we do that without nginx?

Could be possible. but I don't know if requests will be balanced by running just two services.

**Comment by saw-jan** [comment]
Tried in docker-compose setup. both services are getting the request logs. So it might work without nginx
```
storageusers-2-1  | {"level":"info","service":"storage-users","pkg":"rgrpc","traceid":"aa8668aa09b3ac33fc846d5e1147b7dc","data-server":"http://host.docker.internal:9159/data","ref":{"resource_id":{"storage_id":"storage-users-1","opaque_id":"1505e2a2-4a9e-4769-8171-722772b86c0f","space_id":"1505e2a2-4a9e-4769-8171-722772b86c0f"},"path":"./New file (1).txt"},"time":"2024-11-12T07:08:37Z","line":"github.com/cs3org/reva/v2@v2.26.0/internal/grpc/services/storageprovider/storageprovider.go:296","message":"file download"}
storageusers-1-1  | {"level":"info","service":"storage-users","pkg":"rgrpc","traceid":"a40a35729027edafc3d25a817af3548f","data-server":"http://host.docker.internal:9158/data/simple/a850ea6e-2aed-4283-ac84-58f9c54ede7b","fn":"./New file (1).txt","xs":"map[md5:100 unset:1000]","time":"2024-11-12T07:08:41Z","line":"github.com/cs3org/reva/v2@v2.26.0/internal/grpc/services/storageprovider/storageprovider.go:470","message":"file upload"}
storageusers-1-1  | {"level":"info","service":"storage-users","pkg":"rgrpc","traceid":"a40a35729027edafc3d25a817af3548f","data-server":"http://host.docker.internal:9158/data/tus/a850ea6e-2aed-4283-ac84-58f9c54ede7b","fn":"./New file (1).txt","xs":"map[md5:100 unset:1000]","time":"2024-11-12T07:08:41Z","line":"github.com/cs3org/reva/v2@v2.26.0/internal/grpc/services/storageprovider/storageprovider.go:470","message":"file upload"}
storageusers-2-1  | {"level":"info","service":"storage-users","pkg":"rgrpc","traceid":"b04816b03e5b8ad839787e5f76d02970","data-server":"http://host.docker.internal:9159/data","ref":{"resource_id":{"storage_id":"storage-users-1","opaque_id":"1505e2a2-4a9e-4769-8171-722772b86c0f","space_id":"1505e2a2-4a9e-4769-8171-722772b86c0f"},"path":"./New file (1).txt"},"time":"2024-11-12T07:08:43Z","line":"github.com/cs3org/reva/v2@v2.26.0/internal/grpc/services/storageprovider/storageprovider.go:296","message":"file download"}
```

File: .drone.star [hunk 1]
```diff
@@ -263,6 +263,20 @@ config = {
             "tikaNeeded": True,
         },
     },
+    "e2eMultiService": {
+        "testSuites": {
+            "skip": False,
+            "suites": [
+                # "smoke", NOTE: large upload test fails. uncomment when fixed
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply .drone.star:270]: It's only the problem with drone setup. I can run the upload test and it passes on local docker compose setup (multi-service)

[the remark this answers is not in the document]
Comment (saw-jan) [reply .drone.star:270]: https://github.com/owncloud/ocis/issues/7614#issuecomment-2485182434
>I can reproduce the large upload failure with a docker-compose setup.
>When running the upload e2e test from the host machine, the test works fine. But if I run the same test from a docker container (this is what drone does) then the large upload test fails.

[the remark this answers is not in the document]
Comment (saw-jan) [reply .drone.star]: Happy tests: https://drone.owncloud.com/owncloud/ocis/41198/48/14

**Review by ScharfViktor — APPROVED** [review]
cool 👍 great stuff

**Comment by saw-jan** [comment]
I am merging this. Thanks

**Pull request merged**